plugin/src/main/java/org/wildfly/plugins/bootablejar/maven/goals/AbstractBuildBootableJarMojo.java

```diff
@@ -389,12 +399,25 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             throw new MojoExecutionException("Provisioning failed", ex);
         }
 
+        // EE-9
+        String originalLocalRepo = System.getProperty("maven.repo.local");
+        // end EE-9
+
         try {
             // We are forking CLI executions in order to avoid JBoss Modules static references to ModuleLoaders.
             forkCli = Boolean.parseBoolean(pluginOptions.getOrDefault("jboss-fork-embedded", "false"));
             if (forkCli) {
                 getLog().info("CLI executions are done in forked process");
             }
+
+            // EE-9
+            String provisioningRepo = pluginOptions.get("jboss-maven-provisioning-repo");
+            provisioningMavenRepo = provisioningRepo == null ? null : Paths.get(provisioningRepo);
+            if (provisioningMavenRepo != null) {
+                System.setProperty("maven.repo.local", provisioningMavenRepo.toAbsolutePath().toString());
+            }
```

Comment:
I can't get this option to work in the examples. What happens here if a user does `mvn clean install wildfly-jar:package`? Where will the artifact be installed?

Reply:
This option has to be set to the path of an existing local maven repository that contains transformed artifacts, for example add the following to slim example:
Replace the feature-pack to point to the EE9 22 beta one.
<feature-pack-location>wildfly-preview@maven(org.jboss.universe:community-universe)#22.0.0.Beta1-SNAPSHOT</feature-pack-location>

Then add the following options:
<jboss-jakarta-transform-artifacts>false</jboss-jakarta-transform-artifacts>
<jboss-maven-provisioning-repo>/home/jdenise/workspaces/wf19/wildfly-jfdenise/ee-9/feature-pack/target/jakarta-transform-maven-repo/</jboss-maven-provisioning-repo>

That is different from <jboss-maven-repo>target/my-maven-repo</jboss-maven-repo> that is generated during build. When you have the 2 repos in the options, subset of artifacts located in the first one are copied onto the second one.
So you can then start the bootable JAR using a local cache tha tcontains only the subset of artifacts required by the layers you are using:
java -Dmaven.repo.local=/home/jdenise/workspaces/wildfly-jar-maven-plugin/examples/slim/target/my-maven-repo -jar target/slim-bootable.jar

Reply:
One thing I noticed was is it requires to the server to be build in the overridden directory. Maybe that's okay, but it just feels weird to override that property.